cms/lib/xblock/upstream_sync.py

```diff
@@ -96,39 +99,45 @@ def upstream_link(self) -> str | None:
         """
         Link to edit/view upstream block in library.
         """
-        if self.version_available is None or self.upstream_ref is None:
+        if self.version_available is None or self.upstream_key is None:
             return None
-        try:
-            usage_key = LibraryUsageLocatorV2.from_string(self.upstream_ref)
-        except InvalidKeyError:
-            return None
-        return _get_library_xblock_url(usage_key)
+        if isinstance(self.upstream_key, LibraryUsageLocatorV2):
+            return _get_library_xblock_url(self.upstream_key)
+        if isinstance(self.upstream_key, LibraryContainerLocator):
+            return _get_library_container_url(self.upstream_key)
+        return None
 
     def to_json(self) -> dict[str, t.Any]:
         """
         Get an JSON-API-friendly representation of this upstream link.
         """
-        return {
+        data = {
             **asdict(self),
             "ready_to_sync": self.ready_to_sync,
             "upstream_link": self.upstream_link,
         }
+        del data["upstream_key"]  # As JSON (string), this would be redundant with upstream_ref
+        return data
 
     @classmethod
-    def try_get_for_block(cls, downstream: XBlock) -> t.Self:
+    def try_get_for_block(cls, downstream: XBlock, log_error: bool = True) -> t.Self:
         """
         Same as `get_for_block`, but upon failure, sets `.error_message` instead of raising an exception.
         """
         try:
             return cls.get_for_block(downstream)
         except UpstreamLinkException as exc:
-            logger.exception(
-                "Tried to inspect an unsupported, broken, or missing downstream->upstream link: '%s'->'%s'",
-                downstream.usage_key,
-                downstream.upstream,
-            )
+            # Note: if we expect that an upstream may not be set at all (i.e. we're just inspecting a random
+            # unit that may be a regular course unit), we don't want to log this, so set log_error=False then.
+            if log_error:
```

Comment:
Simply viewing a regular unit in Studio (even before this PR) was resulting in a ton of these warnings:

```
upstream_sync.py:133 - Tried to inspect an unsupported, broken, or missing downstream->upstream link: 'block-v1:BradenX+CP200+23+type@vertical+block@9d1d7e5c0b4a42958f7e4174879ec1e2'->'None'
... cms.lib.xblock.upstream_sync.NoUpstream: Content is not linked to a Content Library.
```
So I updated the code to suppress the logging in general. Most things don't have an upstream so it's expected.